src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -136,8 +137,12 @@ class RustProjectSettingsPanel(
 
                 pathToStdlibField.isEditable = !hasRustup
                 pathToStdlibField.setButtonEnabled(!hasRustup)
-                if (stdlibLocation != null && (pathToStdlibField.text.isBlank() || hasRustup)) {
-                    pathToStdlibField.text = stdlibLocation
+                val isWslToolchain = WslDistributionManager.isWslPath(pathToToolchain?.toString().orEmpty())
+                val isWslStdlib = WslDistributionManager.isWslPath(pathToStdlibField.text)
+                if (stdlibLocation != null &&
+                    (pathToStdlibField.text.isBlank() || hasRustup) ||
+                    isWslToolchain != isWslStdlib) {
+                    pathToStdlibField.text = stdlibLocation.orEmpty()
```

Comment:
Let's extract new code related to WSL into something like
```
private fun isStdlibLocationCompatible(toolchainLocation: String, stdlibLocation: String): Boolean {
    ...
}
```
and add explanation comment why we need it